src/plugins/discover/public/application/view_alert/view_alert_route.tsx

```diff
@@ -0,0 +1,203 @@
+  from: string | null;
+  to: string | null;
+  checksum: string | null;
+}
+
+type NonNullableEntry<T> = { [K in keyof T]: NonNullable<T[keyof T]> };
+
+const LEGACY_BASE_ALERT_API_PATH = '/api/alerts';
+const DISCOVER_MAIN_ROUTE = '/';
+
+const displayError = (title: string, errorMessage: string, toastNotifications: ToastsStart) => {
+  toastNotifications.addDanger({
+    title,
+    text: toMountPoint(<MarkdownSimple>{errorMessage}</MarkdownSimple>),
+  });
+};
+
+const displayRuleChangedWarn = (toastNotifications: ToastsStart) => {
+  const warnTitle = i18n.translate('discover.viewAlert.alertRuleChangedWarnTitle', {
+    defaultMessage: 'Alert rule has changed',
+  });
+  const warnDescription = i18n.translate('discover.viewAlert.alertRuleChangedWarnDescription', {
+    defaultMessage: `Displayed documents might not match the documents triggered notification, 
+    since the rule configuration has been changed.`,
+  });
+
+  toastNotifications.addWarning({
+    title: warnTitle,
+    text: toMountPoint(<MarkdownSimple>{warnDescription}</MarkdownSimple>),
+  });
+};
+
+const displayPossibleDocsDiffInfoAlert = (toastNotifications: ToastsStart) => {
+  const infoTitle = i18n.translate('discover.viewAlert.documentsMayVaryInfoTitle', {
+    defaultMessage: 'Displayed documents may vary',
+  });
+  const infoDescription = i18n.translate('discover.viewAlert.documentsMayVaryInfoDescription', {
+    defaultMessage: `The displayed documents might differ from the documents that triggered the alert.
+       Some documents might have been added or deleted.`,
+  });
+
+  toastNotifications.addInfo({
+    title: infoTitle,
+    text: toMountPoint(<MarkdownSimple>{infoDescription}</MarkdownSimple>),
+  });
+};
+
+const getCurrentChecksum = (params: SearchThresholdAlertParams) =>
+  sha256.create().update(JSON.stringify(params)).hex();
+
+const isActualAlert = (queryParams: QueryParams): queryParams is NonNullableEntry<QueryParams> => {
+  return Boolean(queryParams.from && queryParams.to && queryParams.checksum);
+};
+
+export function ViewAlertRoute() {
+  const { core, data, locator, toastNotifications } = useDiscoverServices();
+  const { id } = useParams<{ id: string }>();
+  const history = useHistory();
+  const { search } = useLocation();
+
+  const query = useMemo(() => new URLSearchParams(search), [search]);
+
+  const queryParams: QueryParams = useMemo(
+    () => ({
+      from: query.get('from'),
+      to: query.get('to'),
+      checksum: query.get('checksum'),
+    }),
+    [query]
+  );
+
+  const openConcreteAlert = isActualAlert(queryParams);
+
+  useEffect(() => {
+    const fetchAlert = async () => {
+      try {
+        return await core.http.get<Alert<SearchThresholdAlertParams>>(
+          `${LEGACY_BASE_ALERT_API_PATH}/alert/${id}`
+        );
+      } catch (error) {
+        const errorTitle = i18n.translate('discover.viewAlert.alertRuleFetchErrorTitle', {
+          defaultMessage: 'Alert rule fetch error',
```

Comment:
Could you please provide screenshots of these 3 errors?

Reply:
**Normal flow**

![462C8C4A-F583-40C6-B38B-D6AFBE2DEA15_4_5005_c](https://user-images.githubusercontent.com/39378793/158755195-e2c7ed92-6824-4b94-b2bc-259fb263b878.jpeg)

**When alert config has been changed**

![BF2769A7-6FDC-4FCF-A9DF-8DE5DA9F2467_4_5005_c](https://user-images.githubusercontent.com/39378793/158755416-8a49b6d2-e8c1-4876-827b-1e99537f4377.jpeg)

**Alert was removed in the meantime**

![8CDD01BD-A213-4AD7-9324-34AE1EE2A7A1_4_5005_c](https://user-images.githubusercontent.com/39378793/158755597-ca6b01fe-9697-4fab-aab5-0739dc258dcb.jpeg)

**Data view has been removed in the meantime**

![46FD0B18-D7D5-4E2E-BFA1-A006F4BE36B5_4_5005_c](https://user-images.githubusercontent.com/39378793/158757520-3977dd7d-4141-47dd-959a-123ebf2ae57c.jpeg)

**Index has been removed in the meantime**

![1F42CFAA-BE18-430A-AC76-FF89C649E5D8_4_5005_c](https://user-images.githubusercontent.com/39378793/158759479-11dd908f-4715-4e05-ab4d-1d0a188e45ba.jpeg)

Reply:
For the titles of the two errors, I suggest:

Error fetching alert rule
Error fetching search source

If the last error is in the scope of this PR, I suggest removing the first sentence because the two sentences say the same thing.  So the message would be:

> No indices match "my-index".

Be sure to include an ending period.